plugins/scaffolder/src/next/TemplateWizardPage/TemplateWizardPage.tsx

```diff
@@ -13,13 +13,98 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
+import React from 'react';
+import {
+  Page,
+  Header,
+  Content,
+  Progress,
+  InfoCard,
+  MarkdownContent,
+} from '@backstage/core-components';
 import { FieldExtensionOptions } from '../../extensions';
+import { Navigate, useParams } from 'react-router';
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { errorApiRef, useApi, useRouteRef } from '@backstage/core-plugin-api';
+import { scaffolderApiRef } from '../../api';
+import useAsync from 'react-use/lib/useAsync';
+import { makeStyles } from '@material-ui/core';
+import { Stepper } from './Stepper';
+import { BackstageTheme } from '@backstage/theme';
+import { nextRouteRef } from '../../routes';
 
 export interface TemplateWizardPageProps {
   customFieldExtensions: FieldExtensionOptions<any, any>[];
 }
 
-export const TemplateWizardPage = (_props: TemplateWizardPageProps) => {
-  return null;
+const useStyles = makeStyles<BackstageTheme>(() => ({
+  markdown: {
+    /** to make the styles for React Markdown not leak into the description */
+    '& :first-child': {
+      marginTop: 0,
+    },
+    '& :last-child': {
+      marginBottom: 0,
+    },
+  },
+}));
+
+const useTemplateParameterSchema = (templateRef: string) => {
+  const scaffolderApi = useApi(scaffolderApiRef);
+  const { value, loading, error } = useAsync(
+    () => scaffolderApi.getTemplateParameterSchema(templateRef),
+    [scaffolderApi, templateRef],
+  );
+
+  return { manifest: value, loading, error };
+};
+
+export const TemplateWizardPage = (props: TemplateWizardPageProps) => {
+  const styles = useStyles();
+  const rootRef = useRouteRef(nextRouteRef);
+  const { templateName, namespace } = useParams();
+  const errorApi = useApi(errorApiRef);
+  const { loading, manifest, error } = useTemplateParameterSchema(
+    stringifyEntityRef({
+      kind: 'Template',
+      namespace,
+      name: templateName,
+    }),
+  );
+
+  if (error) {
+    errorApi.post(new Error(`Failed to load template, ${error}`));
+    return <Navigate to={rootRef()} />;
+  }
```

Comment:
technically the post should be in a useEffect so the renderer stays side effect free i guess